src/main/java/com/disraely/ioc/Ioc.java

```diff
@@ -0,0 +1,28 @@
+package com.disraely.ioc;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Ioc {
+    public static Ioc instance;
+    public static Map<String, Object> instances = new HashMap<>();
+
+    public Ioc() {
+    }
+
+    public static Ioc getInstance() {
+        if (instance == null) {
+            instance = new Ioc();
+        }
+        return instance;
+    }
+
+    public void add(String key, Object value) {
+        instances.put(key, value);
```

Comment:
What would happen if the key exists? you will lose the previous instance

Reply:
Resolved!

Reply:
Use `orElseThrow()`

```
var validatedKey =  Optional.ofNullable(key).filter(k -> !instances.containsKey(k)).orElseThrow(...);

instances.put(validatedKey, value);
```

Reply:
Resolved!